cmd/piper_test.go

```diff
@@ -28,6 +28,7 @@ type execCall struct {
 type shellMockRunner struct {
 	dir            string
 	calls          []string
+	shell          []string
```

Comment:
I am surprised. Isn't `shell` the same than `calls`

Reply:
no, shell records the shells used for the call and call records the scripts used:

`RunShell(s string, c string)`

s will be recorded in shell
c will be recorded in calls

Reply:
Is that really needed? Why not re-using the already existing `calls`? I guess there are no cases where we have both at the same time. Would be cool to keep the mock small.

Reply:
yep, function takes two arguments and I want to validate both in the test ...